qiskit_machine_learning/neural_networks/neural_network.py

```diff
@@ -27,12 +28,13 @@ class NeuralNetwork(ABC):
     batched inputs. This is to be implemented by other (quantum) neural networks.
     """
 
-    def __init__(self, num_inputs: int, num_weights: int,
+    def __init__(self, num_inputs: int, num_weights: int, dense: bool,
```

Comment:
Should we define a default for dense so that for most users it works without needing to set this, though that would mean changing the order and moving it to end if output_shape has no meaningful default and must always be given.

Reply:
We have currently 2 implementations, for one it is always dense, for the second both makes sense. I'd like to keep like that for now.